### ISSUE TITLE
"out of the box", npm test fails

### ISSUE THREAD

**Issue by mreinstein** [pixijs/pixi.js#3978]
repro steps:

1. clone pixi repo

2. checkout dev branch

3. `npm install` to get all deps

4. run `npm test` or `npm run test:debug`

5. notice either one fails:

```bash
pixi.js@4.5.1 test /Users/michaelreinstein/wwwroot/pixi.js
floss --path test/index.js

[floss] caught outer exception: Cannot find module '../dist/pixi' Error: Cannot find module '../dist/pixi'
    at Module._resolveFilename (module.js:470:15)
    at Function.Module._resolveFilename (/Users/michaelreinstein/wwwroot/pixi.js/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/michaelreinstein/wwwroot/pixi.js/test/index.js:4:1)
    at Object.<anonymous> (/Users/michaelreinstein/wwwroot/pixi.js/test/index.js:21:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
```

appears there is some kind of pre-build/prep step that is not being run?

**Comment by bigtimebuddy** [comment]
Yup, you need to run `npm run build` before running the tests.

**Comment by mreinstein** [comment]
open to modifying the `test` command?  

Thinking something like this: `"test": "npm run build && floss --path test/index.js",`

**Comment by bigtimebuddy** [comment]
Thanks for the suggestion, but I'm not a huge fan of this idea. Running the tests and building the code are two different workflows. I would be open to having a more verbose error in the test suite if a build file cannot be found, but this is the intended design.

**Comment by mreinstein** [comment]
> Running the tests and building the code are two different workflows. 

They are different workflows, but the test workflow _depends_ on the build workflow.

> the intended design

Maybe so, but the intended design runs counter to the expectations for how tests in npm/node typically work.  In most modules, you can simply clone the code, run `npm i` to get the deps, and then run the standard `npm test` command to validate things work.

**Comment by englercj** [comment]
> They are different workflows, but the test workflow depends on the build workflow.

I agree with this, it is an error to run the tests without building first; seems like we should enforce that invariant. If you want to have a test flow that bypasses building (maybe you are watching in a different console) then that seems like a separate script than the main `npm test` command.

> the intended design runs counter to the expectations for how tests in npm/node typically work.

I again agree with this. If I clone a new repo, and it has `npm test`. I expect to be able to run it without having to run custom scripts manually first.

**Comment by bigtimebuddy** [comment]
Fixed with #3984
